lib/index.js

```diff
@@ -307,28 +307,25 @@ InsightAPI.prototype.blockEventHandler = function(hashBuffer) {
   }
 };
 InsightAPI.prototype.transactionEventHandler = function(txBuffer) {
-  var tx = new Transaction().fromBuffer(txBuffer);
-  var result = this.txController.transformInvTransaction(tx);
-
-  setTimeout(function(result, txBuffer) { // delay tx relay for 900ms to see if we receive transaction lock
-    var hash = dashcore.crypto.Hash.sha256sha256(txBuffer);
-    var id = hash.toString('binary');
-
-    if (this.node.services.dashd.zmqKnownTransactionLocks.get(id)) {
```

Comment:
The lock logic generally didn't change - would that mean that partners will now adapt and subscribe separately for the transactions and locks? As far as I can tell this hack was there to provide a lock field on the transaction, but now there will be no field here, and the reason why it's stopped working is the change in IS Lock message format. Wouldn't users need to listen to two events instead of one in this case?

Reply:
My theory is that partners stopped to listen to the `txlock` subscription, because of the fact that listening to it was crashing insight every half an hour or so (see reason described in previous comment^) That was the initial reason why someone build in the hack to make the txlock detection possible by listening just to the `tx` subscription. This is not how the websocket api is supposed to be used. It is supposed to provide a clear separation between the zmq notifications `pubrawtx` and `pubrawtxlock`. The idea is that if someone was interested in detecting InstantSend, then they would have to listen to both of these websocket channels.